src/stan/io/deserializer.hpp

```diff
@@ -981,6 +981,130 @@ class deserializer {
+   * probability to increment.
+   * @param rows Rows of matrix
+   * @param cows Cows of matrix
+   */
+  template <typename Ret, bool Jacobian, typename LP,
+            require_not_std_vector_t<Ret>* = nullptr,
+            require_matrix_t<Ret>* = nullptr>
+  inline auto read_constrain_stochastic_column(LP& lp, Eigen::Index rows,
+                                               Eigen::Index cols) {
+    using stan::math::stochastic_column_constrain;
+    if (Jacobian) {
+      return stochastic_column_constrain(
+          this->read<conditional_var_val_t<Ret, matrix_t>>(rows - 1, cols), lp);
+    } else {
+      return stochastic_column_constrain(
+          this->read<conditional_var_val_t<Ret, matrix_t>>(rows - 1, cols));
+    }
+  }
+
+  /**
+   * Specialization of \ref read_constrain_stochastic_column for `std::vector`
+   * return types.
+   *
+   * <p>See <code>stan::math::stochastic_column_constrain(T,T&)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam LP Type of log probability.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param lp The reference to the variable holding the log
+   * probability to increment.
+   * @param vecsize The size of the return vector.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * @return Standard vector of matrices transformed to have simplixes along the
+   * columns.
+   */
+  template <typename Ret, bool Jacobian, typename LP, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  inline auto read_constrain_stochastic_column(LP& lp, const size_t vecsize,
+                                               Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(
+          this->read_constrain_stochastic_column<value_type_t<Ret>, Jacobian>(
+              lp, sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Return the next object transformed to a matrix with simplexes along the
+   * rows
+   *
+   * <p>See <code>stan::math::stochastic_row_constrain(T,T&)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam LP Type of log probability.
+   * @param lp The reference to the variable holding the log
+   * probability to increment.
+   * @param rows Rows of matrix
+   * @param cows Cows of matrix
+   */
+  template <typename Ret, bool Jacobian, typename LP,
+            require_not_std_vector_t<Ret>* = nullptr,
+            require_matrix_t<Ret>* = nullptr>
+  inline auto read_constrain_stochastic_row(LP& lp, Eigen::Index rows,
+                                            Eigen::Index cols) {
+    using stan::math::stochastic_row_constrain;
+    if (Jacobian) {
```

Comment:
It looks like there is already an overload in math which checks the template param, can we call that one for consistency? (same for column above)

https://github.com/stan-dev/math/blob/develop/stan/math/prim/constraint/stochastic_row_constrain.hpp#L95-L99